### PR TITLE
DOCS - Reorganize With Diataxis 

### PR DESCRIPTION
**Pull request by lucas-tortora**
# Description of change

- Added default folder structure
- Added default folder structure _README files
- Moved current files to default folder structure
- Updated documentation/README.md
- Updated file imports to new MDX style
- Stylized badges
- Updated sidebar

## Type of change

- [x] Enhancement (a non-breaking change which adds functionality)
- [z] Documentation Fix

## How the change has been tested

Wiki was built locally

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: documentation/README.md [hunk 1]
```diff
@@ -23,14 +23,43 @@ yarn start
...
+You can find guidance and an introduction to the needed tools and syntax on [our wiki](https://wiki.iota.org/participate/contribute-to-wiki/for_devs/developer_guide)
 
-Example:
 
-```console
-{@import ../../../../bindings/wasm/docs/api-reference.md}
-```
+## Default Structure
+
+IOTA projects usually house their Wiki documentation within the documentation folder of any project. This
```
Comment (aleksei-korolev) [inline documentation/README.md]:
```suggestion
IOTA projects usually house their Wiki documentation in the `documentation` folder within their repositories. This
```

File: documentation/README.md [hunk 2]
```diff
@@ -23,14 +23,43 @@ yarn start
...
+template provides a set structure to separate concerns based on the [Diataxis framework](https://diataxis.fr/), with
```
Comment (aleksei-korolev) [inline documentation/README.md]:
```suggestion
template provides a structure to separate reader concerns, applying the [Diataxis framework](https://diataxis.fr/) with
```

File: documentation/README.md [hunk 3]
```diff
@@ -23,14 +23,43 @@ yarn start
...
+some minor modifications.
+
+You do not need to use all of these sections, so please feel free to delete any which do not suit your needs.
```
Comment (aleksei-korolev) [inline documentation/README.md]:
```suggestion
You do not have to use all of these sections, so please feel free to delete any which do not suit your needs.
```

File: documentation/README.md [hunk 4]
```diff
@@ -23,14 +23,43 @@ yarn start
...
+
+### Getting Started
+
+The [Getting Started folder](documentation/getting_started/_README.md) should be the first section in your documentation. It should give the user a high-level overview of the project, required prior knowledge, prerequisites, and ideally a quick setup guide, or "hello world".
+
+### How
```
Comment (aleksei-korolev) [inline documentation/README.md]: Shouldn't it be How To or How Tos? Just double checking

File: documentation/README.md [hunk 5]
```diff
@@ -23,14 +23,43 @@ yarn start
...
+
+The [How Tos folder](documentation/how_tos/_README.md) should only address concrete examples, or [how-to guides](https://diataxis.fr/how-to-guides/), which are **goal oriented**.
```
Comment (aleksei-korolev) [inline documentation/README.md]:
```suggestion
The [How Tos folder](documentation/how_tos/_README.md) should only address concrete examples, or [how-to guides](https://diataxis.fr/how-to-guides/), which are **goal-oriented**.
```

File: documentation/README.md [hunk 6]
```diff
@@ -23,14 +23,43 @@ yarn start
...
+
+### Tutorials
+
+The [Tutorials folder](documentation/tutorials/_README.md) should contain articles which guide the user step by step through a series of how-tos with the relevant explanations to achieve a project or real world use case. [Tutorials](https://diataxis.fr/tutorials/) are **learning-oriented**.
+
+### Key Concepts
+
+The [Key concepts folder](documentation/key_concepts/_README.md) revolves around [explanations](https://diataxis.fr/explanation/), and are therefore **understanding oriented**.
```
Comment (aleksei-korolev) [inline documentation/README.md]:
```suggestion
The [Key concepts folder](documentation/key_concepts/_README.md) revolves around [explanations](https://diataxis.fr/explanation/), and are therefore **understanding-oriented**.
```

File: documentation/docs/how_tos/_README.mdx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  - How Tos
+  - action oriented
+  - how
+---
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# How To Files
+
+How Tos are analogous to [how-tos in Diataxis](https://diataxis.fr/explanation/). You should use this folder to give in depth guides on how to complete concrete single tasks. For example:
```
Comment (aleksei-korolev) [inline documentation/docs/how_tos/_README.mdx]:
```suggestion
How Tos are the same as [how-tos in Diataxis](https://diataxis.fr/how-to-guides/). You should use this folder to give in depth guides on how to complete concrete single tasks. For example:
```

File: documentation/docs/how_tos/_README.mdx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+- Connect to a Server
+- Send a Message
+- Create a Transaction
+- Connect to Stronghold
+
+## Action Oriented
```
Comment (aleksei-korolev) [inline documentation/docs/how_tos/_README.mdx]:
```suggestion
## Action-Oriented
```

File: documentation/docs/how_tos/_README.mdx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+How To files should be **action oriented**. This means that you should first think of what you want the user to achieve
```
Comment (aleksei-korolev) [inline documentation/docs/how_tos/_README.mdx]:
```suggestion
How To files should be **action-oriented**. This means that you should first think of what you want the user to achieve
```

**Review by aleksei-korolev — CHANGES_REQUESTED** [review]

**Review by felsweg-iota — APPROVED** [review]
Thanks @lucas-tortora for providing the wiki changes

File: documentation/README.md (hunk 4 above)
Comment (lucas-tortora) [reply documentation/README.md]:
```suggestion
### How Tos
```

File: documentation/docs/getting_started.md [hunk 10]
```diff
@@ -38,15 +39,16 @@ If you want to get involved in discussions about this library, or you're looking
...
-3. **The Specification**: detailed explanation of requirements and functionality
-4. **Retrospective**: a look at the evolution of this project
-5. **Contribute**: how you can participate in the Stronghold software development
-6. **Get in touch**: join the community and become part of the X-Team
+1. **[Overview](reference/overview.md)**: detailed overview of the project
+2. **[Structure](reference/structure/overview.md)**: explains the layout of the individual crates and systems
+3. **[The Specification](reference/specs/overview.md)**: detailed explanation of requirements and functionality
+4. **[Retrospective](key_concepts/retrospective.md)**: a look at the evolution of this project
+5. **[Contribute](contribute.md)**: how you can participate in the Stronghold software development
+6. **[Get in touch](contribute#join-the-iota-stronghold-initiative)**: join the community and become part of the X-Team
```
Comment (aleksei-korolev) [inline documentation/docs/getting_started.md:47]: Is it still relevant with the new ToC structure?

**Review by aleksei-korolev — APPROVED** [review]

File: documentation/docs/getting_started.md (hunk 10 above)
Comment (lucas-tortora) [reply documentation/docs/getting_started.md:47]: Not particularly, but this PR should address structural changes for the most part, and of course not break anything.  We can update content in subsequent PRs with Stronghold 1.0

File: documentation/README.md [hunk 11]
```diff
@@ -23,14 +23,43 @@ yarn start
 
 This command starts a local, wiki themed development server and opens up a browser window. Most changes are reflected live without having to restart the server.
 
-## Including .md file
+## Features And Tools
```
Comment (Dr-Electron) [inline documentation/README.md:26]: Do we want to keep this section for now or replace it with the style guide from the wiki directly and update that one in the next days?

Comment (Dr-Electron) [reply documentation/README.md:26]: We decided on removing the content once the wiki is ready

**Review by Dr-Electron — APPROVED** [review]

**Pull request merged**